gulpfile.js

```diff
@@ -0,0 +1,381 @@
+ * DevApp Build Tasks.
+ */
+gulp.task(':build:e2eapp:ts', [':build:components:ts'], createTsBuildTask({ tsConfigPath: e2eAppDir }));
+gulp.task(':build:e2eapp:scss', [':build:components:scss'], createSassBuildTask({
+  dest: outDir,
+  root: e2eAppDir,
+  // Change this once we have a better strategy for releasing SCSS files.
+  includePaths: [
+    path.join(componentsDir, 'core/style'),
+    componentsDir
+  ]
+}));
+gulp.task(':build:e2eapp:assets', function() {
+  return gulp.src(path.join(e2eAppDir, '**/*'))
+    .pipe(gulp.dest(outDir));
+});
+
+gulp.task('build:e2eapp', [
+  'build:components',
+  ':build:vendor',
+  ':build:e2eapp:ts',
+  ':build:e2eapp:scss',
+  ':build:e2eapp:assets'
+]);
+
+
+/***************************************************************************************************
+ * Global tasks.
+ */
+gulp.task('default', ['help']);
+
+gulp.task('help', function() {
+  const tasks = Object.keys(gulp.tasks)
+    .filter(x => !x.startsWith(':'))
+    .filter(x => !x.startsWith('ci:'))
+    .filter(x => x != 'default')
+    .sort();
+
+  console.log(`\nHere's a list of supported tasks:\n   `, tasks.join('\n    '));
+  console.log(`\nYou're probably looking for "test" or "serve:devapp".\n\n`);
+});
+
+gulp.task('build', ['build:devapp']);
+
+gulp.task('clean', function() {
+  return gulp.src('dist', { read: false })
+    .pipe(gulpClean());
+});
+gulp.task(':clean:spec', function() {
+  return gulp.src('dist/**/*.spec.*', { read: false })
+    .pipe(gulpClean());
+});
+gulp.task(':clean:assets', function() {
+  return gulp.src('dist/**/*+(.html|.css)', { read: false })
+    .pipe(gulpClean());
+});
+
+gulp.task('lint', createExecTask('tslint', ['-c', 'tslint.json', 'src/**/*.ts']));
+gulp.task('stylelint', createExecTask(
+  'stylelint', ['src/**/*.scss', '--config', 'stylelint-config.json', '--syntax', 'scss']
+));
+
+/***************************************************************************************************
+ * Watch Tasks.
+ */
+gulp.task(':watch:components', ['build:components'], function() {
+  return gulp.watch(path.join(componentsDir, '**/*'), ['build:components']);
+});
+
+
+/***************************************************************************************************
+ * Serve Tasks.
+ */
+gulp.task('serve:devapp', ['build:devapp'], function() {
+  const stream = gulp.src('dist')
+    .pipe(gulpServer({
+      livereload: true,
+      fallback: 'index.html',
+      port: 4200
+    }));
+
+  gulp.watch(path.join(componentsDir, '**/*.ts'), [':build:components:ts']);
+  gulp.watch(path.join(componentsDir, '**/*.scss'), [':build:components:scss']);
+  gulp.watch(path.join(componentsDir, '**/*.html'), [':build:components:assets']);
+  gulp.watch(path.join(devAppDir, '**/*.ts'), [':build:devapp:ts']);
+  gulp.watch(path.join(devAppDir, '**/*.scss'), [':build:devapp:scss']);
+  gulp.watch(path.join(devAppDir, '**/*.html'), [':build:devapp:assets']);
```

Comment:
The `stream` watches and serves the `dist` folder itself, and `watch` runs tasks if a file was changed. When a file change, `build:components:XXX` will run and update the file in `dist`. The livereload server will see the file change and trigger a reload.

So, basically, indirectly.